unity/unity-embed-host/CoreCLRHost.cs

```diff
@@ -14,11 +18,18 @@ static unsafe partial class CoreCLRHost
 {
     static ALCWrapper alcWrapper;
     static FieldInfo assemblyHandleField;
+    private static Dictionary<Assembly, AssemblyPair> m_assemblies = new Dictionary<Assembly, AssemblyPair>();
+
+    private class AssemblyPair
```

Comment:
Nit, I would make this a readonly struct or if keeping a class make the fields readonly.